src/parameters/add_parameters.jl

```diff
@@ -204,28 +219,58 @@ function _add_time_series_parameters!(
     if !(ts_type <: Union{PSY.AbstractDeterministic, PSY.StaticTimeSeries})
         error("add_parameters! for TimeSeriesParameter is not compatible with $ts_type")
     end
-    time_steps = get_time_steps(container)
-    ts_name = get_time_series_names(model)[T]
-    time_series_mult_id = _create_time_series_multiplier_index(model, T)
 
-    @debug "adding" T D ts_name ts_type time_series_mult_id _group =
-        LOG_GROUP_OPTIMIZATION_CONTAINER
+    time_steps = get_time_steps(container)
+    # TODO: Temporary workaround to get the name where we assume all the names are the same accross devices.
+    ts_name = _get_time_series_name(T(), first(devices), model)
 
     device_names = String[]
+    devices_with_time_series = D[]
     initial_values = Dict{String, AbstractArray}()
-    for device in devices
+
+    @debug "adding" T D ts_name ts_type _group = LOG_GROUP_OPTIMIZATION_CONTAINER
+
+    for device::D in devices
         if !PSY.has_time_series(device, ts_type, ts_name)
-            @debug "skipped time series for $D, $(PSY.get_name(device))"
+            @info "Time series $(ts_type):$(ts_name) for $D, $(PSY.get_name(device)) not found skipping parameter addition."
             continue
         end
         push!(device_names, PSY.get_name(device))
+        push!(devices_with_time_series, device)
         ts_uuid = string(IS.get_time_series_uuid(ts_type, device, ts_name))
         if !(ts_uuid in keys(initial_values))
             initial_values[ts_uuid] =
                 get_time_series_initial_values!(container, ts_type, device, ts_name)
+            _check_dynamic_branch_rating_ts(initial_values[ts_uuid], param, device, model)
         end
     end
 
+    #=
+    # NOTE this is always the case for "normal" time series, but it is currently not enforced in PSY for MBC time series.
+    # TODO decide whether this is an acceptable restriction or whether we need to support multiple time series names
+    # JD: Yes, the restriction are that the names for this has to be unique as they are specified from the model attributes
```

Comment:
Not for MBC, right? They're specified by the `TimeSeriesKey`s in the MBC itself, which is per-component